subsys/bluetooth/mesh/gen_plvl_srv.c

```diff
@@ -174,6 +171,10 @@ static void plvl_set(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
 
 	if (!tid_check_and_update(&srv->tid, tid, ctx)) {
 		change_lvl(srv, ctx, &set, &status);
+
+		if (IS_ENABLED(CONFIG_BT_MESH_SCENE_SRV)) {
+			bt_mesh_scene_invalidate(&srv->scene);
+		}
```

Comment:
Generic Level Server invalidates scene on Set/Delta Set/Move Set. I tried to follow approach where it is model's responsibility to invalidate a scene if it receives a message that change state stored with scene. Then scene is always invalidated when the extended model calls its handlers.

The only downside of this as I see is that not all models holds scene entry (e.g., Hue and Saturation servers) so that to invalidate the scene I need to use scene entry of the extended model instance:`bt_mesh_scene_invalidate(&srv->lvl.scene)`.

Reply:
Good point, I didn't consider this.